### ISSUE TITLE
Fatal Error when KerasModelImport , problematic frame at libnd4jcpu.so

### ISSUE THREAD

**Issue by kunhui123** [deeplearning4j/deeplearning4j#9910]
#### Issue Description
Recently was trying to bump up the DL4J version to "1.0.0-M2.1".
The same code was working fine at earlier version like "1.0.0-M1.1" and it now having issue on the "1.0.0-M2.1".

The issue occur by just importing model (.h5) train using Keras as code below.
` val model = KerasModelImport.importKerasSequentialModelAndWeights(modelFile.getAbsolutePath, false)
`
Please describe our issue, along with:
- expected behavior : The model should able to import into DL4J's program without error.
- encountered behavior : My machine was freeze/hang until it receive Fatal Error message as below.

`00:21:50.586 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.factory.Nd4jBackend - Loaded [CpuBackend] backend`
`00:21:51.288 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.nativeblas.NativeOpsHolder - Number of threads used for linear algebra: 2`
`00:21:51.296 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.cpu.nativecpu.CpuNDArrayFactory - Binary level AVX/AVX2 optimization level AVX/AVX2`
`00:21:51.301 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.nativeblas.Nd4jBlas - Number of threads used for OpenMP BLAS: 2`
`00:21:51.308 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Backend used: [CPU]; OS: [Linux]`
`00:21:51.308 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Cores: [4]; Memory: [3.9GB];`
`00:21:51.308 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Blas vendor: [OPENBLAS]`
`00:21:51.313 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.nd4j.linalg.cpu.nativecpu.CpuBackend - Backend `
`build information:`
 `GCC: "7.5.0"`
`STD version: 201103L`
`DEFAULT_ENGINE: samediff::ENGINE_CPU`
`HAVE_FLATBUFFERS`
`HAVE_OPENBLAS`
`00:21:53.303 [pool-1-thread-1-ScalaTest-running-Test_Sentiment] INFO org.deeplearning4j.nn.multilayer.MultiLayerNetwork - Starting MultiLayerNetwork with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE]`

**PC start freeze at this point and after a long time it crash and Fatal Error log (eg: hs_err_pid18909.log) was produced as below.**

`A fatal error has been detected by the Java Runtime Environment:`
`SIGSEGV (0xb) at pc=0x00007f45fc12dd15, pid=18528, tid=18556`

`JRE version: Java(TM) SE Runtime Environment (19.0.1+10) (build 19.0.1+10-21)`
`Java VM: Java HotSpot(TM) 64-Bit Server VM (19.0.1+10-21, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)`
`Problematic frame:`
`C  [libnd4jcpu.so+0x16d8d15]  sd::NDArray::getOffset(long long) const+0x25`

#### Version Information
Please indicate relevant versions, including, if relevant:

* Deeplearning4j version : "1.0.0-M2.1"
* Platform information (OS, etc) : Ubuntu 18.04.6 LTS
* CUDA version, if used : N/A (CpuBackend was used)
* NVIDIA driver version, if in use  : N/A (CpuBackend was used)

#### Additional Information
Full Fatal Error log.
https://gist.github.com/kunhui123/8099f143bedf63821ed676e9ddb29f81

Although not all Keras model fail to import, but for the 2 model that I tested and failed with the fatal error above are share in the as structure and Google Drive link below.
https://drive.google.com/drive/folders/16QA7g96DvxOdXGLOYMy6wrGIhS1IdYyl?usp=sharing

`model = Sequential()`
`model.add(Embedding(len(weight_matrix), EMBEDDING_DIM, weights=[weight_matrix], input_length=max_words, trainable=False))`
`model.add(Bidirectional(LSTM(128, dropout=0.2, recurrent_dropout=0.2)))`
` model.add(Dense(512, activation='relu'))`
` model.add(Dropout(0.50))`
` model.add(Dense(10, activation='softmax'))`
`model.compile(loss='categorical_crossentropy',optimizer='adam', metrics=['accuracy'])`

**Comment by kunhui123** [comment]
HI @agibsonccc , any chance to take a look on this issue ?
In additional , the sample model was build from the article on medium below. 
https://medium.com/analytics-vidhya/sentiment-analysis-for-text-with-deep-learning-2f0a0c6472b5

I hope that I have provide enough info for you to replicate the issue above.

**Comment by agibsonccc** [comment]
@kunhui123  just to make sure it's not already fixed could you try snapshots first? https://deeplearning4j.konduit.ai/multi-project/explanation/snapshots if not sure!

**Comment by kunhui123** [comment]
@agibsonccc I tried to pull the snapshots from the "oss.sonatype.org/content/repositories/snapshots" and it doesn't work.
I believe the jars at snapshots repositories was outdated since the latest & last modified date those jars "Tue Dec 21 06:51:02 UTC 2021".

Alternative, I had pull from latest "master" branch (Feb 13, 2023) from this github and build these "snapshots" jar from sources. But unfortunate, it still no working either.

**Comment by agibsonccc** [comment]
@kunhui123 That's the wrong oss.sonatype. Apologies, I updated the snapshots docs for the upcoming release version but not the current stable. I've since fixed that: https://deeplearning4j.konduit.ai/multi-project/explanation/snapshots

Please try s01.oss.sonatype.org and try again.  The short of it is a while back we migrated to a new host for snapshots and needed to update some of the examples and docs.

**Comment by kunhui123** [comment]
@agibsonccc" I have already retrieved the latest JAR file from "s01.oss.sonatype.org" , as I check the jar was dated "xxx:1.0.0-20230208.214010-2593.jar".
The issue still persist and my model still freeze and at import stage.
`val model = KerasModelImport.importKerasSequentialModelAndWeights(modelFile.getAbsolutePath, false)
`

**Comment by agibsonccc** [comment]
@kunhui123  ah thank you! Apologies I misunderstood "failure" from you. I thought you failed to build from source. Thank you for confirming!

**Comment by agibsonccc** [comment]
@kunhui123  I tried both models in your .h5 and it worked fine. Unfortunately I still can't reproduce this. Maybe your snapshots weren't up to date? If it helps native crashes recently have been fixed related to shape buffers. If you're running in a multi threaded environment that maybe your issue. 

You're welcome to try the snapshots again or ensure you're not running the import in a multi threaded environment (eg: use synchronized) 

In light of this I'm closing this. There shouldn't be anything model import specific here. Feel free to reopen if you think you have a problem here. I'll get notifications if you comment. Please do let me know what you decide to do.
